### PR TITLE
[wrangler] fix: show URL when browser cannot be opened in headless/container environments

### LINKED ISSUE #14023: Missing file or directory: xdg-open
### What versions & operating system are you using?

  System:
    OS: Linux 6.8 Ubuntu 24.04.1 LTS 24.04.1 LTS (Noble Numbat)
    CPU: (12) x64 AMD Ryzen 5 3600X 6-Core Processor
    Memory: 17.27 GB / 31.23 GB
    Container: Yes
    Shell: 5.2.21 - /bin/bash
  Binaries:
    Node: 23.6.0 - /home/ubuntu/.nvm/versions/node/v23.6.0/bin/node
    npm: 10.9.2 - /home/ubuntu/.nvm/versions/node/v23.6.0/bin/npm
    pnpm: 9.15.9 - /home/ubuntu/.nvm/versions/node/v23.6.0/bin/pnpm

### Please provide a link to a minimal reproduction

n/a

### Describe the Bug

When deploying using Wrangler@latest, it tries to login using the browser. If this is done in a headless environment like a server, it will fail and crash Wrangler:

```shell
✘ [ERROR] A file or directory could not be found.

  
  Missing file or directory: xdg-open
  
  This is typically caused by:
    - The file or directory does not exist
    - A typo in the file path
    - The file was moved or deleted
  
  Please check the file path and try again.
```

### Please provide any relevant error logs

_No response_

### PR DESCRIPTION
**Pull request by matingathani**
Fixes #14023.

On Linux systems without `xdg-open` installed (headless servers, Docker containers, StackBlitz), the `open` npm package resolves the path to its bundled `xdg-open` script relative to `__dirname`. When wrangler is bundled into `wrangler-dist/cli.js`, `__dirname` points at the dist directory where the shell script doesn't exist, so `open()` rejects with ENOENT. This error propagated to the generic ENOENT handler in `handle-errors.ts`, which displayed a confusing "Missing file or directory: xdg-open" message with no actionable guidance.

**Fix**: `openInBrowser` now catches ENOENT errors — both from the `open()` promise rejection and from the child process `error` event — and logs a clear warning with the full URL so users can copy-paste it into a browser manually.

```
⚠ [WARNING] Failed to open a browser automatically. You may need to install `xdg-utils`
  (e.g. `apt install xdg-utils`) or set a default browser.
  Please visit the following URL in your browser:
  https://dash.cloudflare.com/oauth2/auth?...
```

---

- Tests
  - [x] Tests included/updated
- Public documentation
  - [x] Documentation not necessary because: the fix is a UX improvement for an edge case — no new config options or commands added

### CODE REVIEW

**Comment by workers-devprod** [comment]
Codeowners approval required for this PR:
- ✅ @cloudflare/wrangler

<details><summary>Show detailed file reviewers</summary>


</details>

File: packages/wrangler/src/open-in-browser.ts [hunk 1]
```diff
@@ -2,16 +2,36 @@ import open from "open";
...
+	}
+}
+
+function handleBrowserOpenError(url: string, err: unknown): void {
+	const hint =
+		process.platform === "linux"
+			? " You may need to install `xdg-utils` (e.g. `apt install xdg-utils`) or set a default browser."
+			: "";
+	logger.warn(
+		`Failed to open a browser automatically.${hint}\n` +
```
[the remark this answers is not in the document]
Comment (matingathani) [reply packages/wrangler/src/open-in-browser.ts:34]: Fixed in the latest commit. The Linux xdg-utils hint is now gated on `isEnoent` — it only appears when the error code is `ENOENT` (i.e. the opener binary wasn't found). Non-ENOENT child-process errors still warn + print the URL, but omit the irrelevant install hint.

File: packages/wrangler/src/open-in-browser.ts [hunk 2]
```diff
@@ -2,16 +2,36 @@ import open from "open";
...
+			`Please visit the following URL in your browser:\n${url}`
+	);
+	logger.debug(String(err));
```
[the remark this answers is not in the document]
Comment (matingathani) [reply packages/wrangler/src/open-in-browser.ts]: Fixed. `logger.debug` now receives `err.stack ?? String(err)` for `Error` instances, preserving the full stack trace. Non-Error values fall back to `String(err)` as before.

**Review by dario-piotrowicz — APPROVED** [review]
LGTM, thanks for the PR @matingathani 👍

**Review by workers-devprod — APPROVED** [review]
Codeowners reviews satisfied
